metricq/interval_source.py

```diff
@@ -50,7 +50,15 @@ async def task(self):
         self._interval_task_stop_future = self.event_loop.create_future()
         deadline = Timestamp.now()
         while True:
-            await self.update()
+            try:
+                await self.update()
+            except MetricSendError as e:
+                # This is a "normal" case, when we lost the connection.
+                # During the reconnection phase, we need to save the task from
+                # being cancelled.
+                logger.debug("Failed to send metric value: {}", e)
+                pass
```

Comment:
`pass` is redundant

Reply:
> I understand that the `loop` parameter is _deprecated_. But are we sure it is _not necessary_?

The [holy SO](https://stackoverflow.com/questions/60312374/what-are-all-these-deprecated-loop-parameters-in-asyncio) says so.

Reply:
> > I understand that the `loop` parameter is _deprecated_. But are we sure it is _not necessary_?
> 
> The [holy SO](https://stackoverflow.com/questions/60312374/what-are-all-these-deprecated-loop-parameters-in-asyncio) says so.

Then we should still keep the `asyncio.Event()` creation in `start` rather than `__init__` because it runs `get_event_loop()` in `Event.__init__`. and `agent.__init__` may be called outside of the eventual event loop.

Reply:
Turns out, we don't need that loop parameter change here. We can deal with that issue later.